pretrain/mining.py

```diff
@@ -39,129 +40,160 @@ def model_path(base_dir: str, run_id: str) -> str:
-        model.save_pretrained(
-            save_directory=model_dir,
-            safe_serialization=True,
-        )
-
-    def load_gpt2_model(self, model_file: str) -> PreTrainedModel:
-        """For loading GPT2 models from the previous version of this subnet."""
-        model = pt.model.get_model()
-        load_model(model, model_file)
-        return model
-
-    def load_local_model(self, model_dir: str) -> PreTrainedModel:
-        """Loads a model from a directory."""
-        return AutoModelForCausalLM.from_pretrained(
-            pretrained_model_name_or_path=model_dir,
-            local_files_only=True,
-            use_safetensors=True,
-        )
-
-    async def load_remote_model(
-        self, uid: int, metagraph: bt.metagraph, download_dir: str
-    ) -> PreTrainedModel:
-        """Loads the model currently being advertised by the Miner with the given UID.
-
-        Args:
-            uid (int): The UID of the Miner who's model should be downloaded.
-            metagraph (bt.metagraph): The metagraph of the current subtensor.
-            download_dir (str): The directory to download the model to.
-        """
-        hotkey = metagraph.hotkeys[uid]
-        model_metadata = await self.model_metadata_store.retrieve_model_metadata(hotkey)
-        if not model_metadata:
-            raise ValueError(f"No model metadata found for miner {uid}")
-
-        model: Model = await self.remote_model_store.download_model(
-            model_metadata.id, download_dir
-        )
-        return model.pt_model
-
-    async def push(self, model: PreTrainedModel, retry_delay_secs: int = 60):
-        """Pushes the model to Hugging Face and publishes it on the chain for evaluation by validators."""
-        bt.logging.info("Pushing model")
-
-        # First upload the model to HuggingFace.
-        model_id = ModelId(namespace=self.hf_repo_namespace, name=self.hf_repo_name)
-        model_id = await self.remote_model_store.upload_model(
-            Model(id=model_id, pt_model=model)
-        )
-
-        bt.logging.success(
-            f"Uploaded model to hugging face. Now committing to the chain with model_id: {model_id}"
-        )
-
-        # We can only commit to the chain every 20 minutes, so run this in a loop, until
-        # successful.
-        while True:
-            try:
-                await self.model_metadata_store.store_model_metadata(
-                    self.wallet.hotkey.ss58_address, model_id
-                )
 
-                bt.logging.info(
-                    "Wrote model metadata to the chain. Checking we can read it back..."
+    # First upload the model to HuggingFace.
+    namespace, name = utils.validate_hf_repo_id(repo)
+    model_id = ModelId(namespace=namespace, name=name)
+    model_id = await remote_model_store.upload_model(Model(id=model_id, pt_model=model))
+
+    bt.logging.success(
+        f"Uploaded model to hugging face. Now committing to the chain with model_id: {model_id}"
```

Comment:
nit: We assume that the remote repo is indeed hugging face in the logs and validating the repo and such. I don't think we need to change here but something to consider when we add more.

Same below for the chain for metadata.